src/sinks/elasticsearch.rs

```diff
@@ -64,8 +66,16 @@ pub struct ElasticSearchConfig {
 
     pub aws: Option<RegionOrEndpoint>,
     pub tls: Option<TlsOptions>,
-    #[serde(default)]
-    pub bulk_action: BulkAction,
+    pub bulk_action: Option<BulkAction>,
+}
+
+impl ElasticSearchConfig {
+    fn bulk_action(&self) -> BulkAction {
+        match self.mode {
+            ElasticSearchMode::Regular => self.bulk_action.clone().unwrap_or(BulkAction::Index),
```

Comment:
I might just `derive(Copy)` on the enum since it has no fields to avoid needing to clone here.